src/mem/cache/tags/partitioning_policies/partitionFieldsExtension.hh

```diff
@@ -0,0 +1,90 @@
+ * terms below provided that you ensure that this notice is replicated
+ * unmodified and in its entirety in all distributions of the software,
+ * modified or unmodified, in source code or in binary form.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met: redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer;
+ * redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution;
+ * neither the name of the copyright holders nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "base/extensible.hh"
+#include "mem/request.hh"
+
+namespace gem5
+{
+
+namespace partitioning_policy
+{
+
+static const uint64_t DEFAULT_PARTITION_ID = 0;
+static const uint64_t DEFAULT_PARTITION_MONITORING_ID = 0;
+
+class PartitionFieldExtention : public Extension<Request,
+                                                 PartitionFieldExtention>
+{
+    public:
+        PartitionFieldExtention(){};
+        std::unique_ptr<ExtensionBase> clone() const override
+            {return std::make_unique<PartitionFieldExtention>(*this);};
+
+        // getters for partition_ID
+        auto getPartitionID() {return this->partition_ID;}
+        auto partitionID()    {return this->partition_ID;}
+
+        // getters for partition_monitoring_ID
+        auto getPartitionMonitoringID()
+            {return this->partition_monitoring_ID;}
+        auto partitionMonitoringID()
+            {return this->partition_monitoring_ID;}
+
+        // setters
+        void setPartitionID(uint64_t id)
+            {this->partition_ID = id;}
+        void setPartitionMonitoringID(uint64_t id)
+            {this->partition_monitoring_ID = id;}
+
+    private:
+        uint64_t partition_ID=DEFAULT_PARTITION_ID;
```

Comment:
Member variables are camel case outside ruby

Reply:
Noted; Formatting fixed in [commit 4d6b3a6](https://github.com/gem5/gem5/pull/765/commits/4d6b3a60d88345d1585d0c1b49dcd4501f39a510)